lib/block-supports/layout.php

```diff
@@ -63,15 +63,21 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 
 	$style = '';
 	if ( $content_size || $wide_size ) {
-		$style  = ".wp-container-$id > * {";
-		$style .= 'max-width: ' . esc_html( $all_max_width_value ) . ';';
-		$style .= 'margin-left: auto !important;';
-		$style .= 'margin-right: auto !important;';
+
+		$style  = ".wp-container-$id {";
+		$style .= 'display: flex;';
+		$style .= 'flex-flow: column;';
+		$style .= 'align-items: center;';
+		$style .= '}';
+
+		$style .= ".wp-container-$id > * {";
+		$style .= 'box-sizing: border-box;';
+		$style .= 'width: ' . esc_html( $all_max_width_value ) . ';';
 		$style .= '}';
 
-		$style .= ".wp-container-$id > .alignwide { max-width: " . esc_html( $wide_max_width_value ) . ';}';
+		$style .= ".wp-container-$id > .alignwide { width: " . esc_html( $wide_max_width_value ) . ';}';
```

Comment:
This doesn't seem to be responsive, it doesn't adapt to the screen size if it's bigger than the width. (thus the previous usage of max-width)

Reply:
We went the grid way and it seems to be working fine responsively now